server/modules/user/repository/userRepository.js

```diff
@@ -126,6 +127,19 @@ const deleteUsersPrefsSurvey = async (surveyId, client = db) => {
 `)
 }
 
+/**
+ * Sets survey cycle user pref to Survey.cycleOneKey if the preferred cycle is among the specified (deleted) ones
+ */
+const deleteUsersPrefsCycleDeleted = async (surveyId, cycleKeysDeleted, client = db) => {
```

Comment:
Maybe this method should just be ```deleteUsersPrefsSurveyCycle = async (surveyId, cycles, client = db) ```

Reply:
or even better, resetUsersPrefsSurveyCycle